test/statistics/client_statistics_test.go

```diff
@@ -229,6 +231,58 @@ func TestClientStatisticsContent(t *testing.T) {
 
 }
 
+func TestClientStatisticsContentChanges(t *testing.T) {
+	cluster, _ := remoteController.CreateCluster("", test.DefaultServerConfig)
+	remoteController.StartMember(cluster.ID)
+	defer remoteController.ShutdownCluster(cluster.ID)
+
+	config := hazelcast.NewConfig()
+	config.SetProperty(property.StatisticsEnabled.Name(), "true")
+	config.SetProperty(property.StatisticsPeriodSeconds.Name(), "1")
+
+	client, _ := hazelcast.NewClientWithConfig(config)
+	defer client.Shutdown()
+	var stats string
+	test.AssertEventually(t, func() bool {
+		stats = GetClientStatsFromServer(t, cluster.ID)
+		return len(stats) > 0
+	})
+
+	runTimeMetrics := getRuntimeMetrics(stats)
+	test.AssertEventually(t, func() bool {
+		nextStats := GetClientStatsFromServer(t, cluster.ID)
+		if stats == nextStats {
+			return false
+		}
+		nextRuntimeMetrics := getRuntimeMetrics(nextStats)
+		return isRuntimeMetricsDifferent(runTimeMetrics, nextRuntimeMetrics)
+	})
+
+}
+
+func isRuntimeMetricsDifferent(runtimeMetrics []string, nextRuntimeMetrics []string) bool {
+	isDifferent := false
+	for index, nextRuntimeMetric := range nextRuntimeMetrics {
```

Comment:
Should you first check the number of stats in the metrics returned before one-by-one comparison?